lib/performance_data/statistics.rb

```diff
@@ -0,0 +1,216 @@
+require 'json'
+require 'gds_api/performance_platform/data_out'
+
+module PerformanceData
+  class Statistics
+    attr_accessor :data_out, :content, :slug, :part_urls, :document_type, :is_multipart, :performance_data
+
+    def initialize(content, slug)
+      @data_out = GdsApi::PerformancePlatform::DataOut.new("http://www.performance.service.gov.uk")
+      @content = content
+      @slug = slug
+      @part_urls = get_part_urls(content, slug)
+      @document_type = @content["document_type"] || @content["format"]
+      @is_multipart = @part_urls.any? || (document_type == 'smart-answer')
+    end
+
+    def performance_data
+      @performance_data ||= {
+        'page_views': page_views,
+        'problem_reports': problem_reports,
+        'searches': searches,
+        'search_terms': search_terms,
```

Comment:
As this method is only ever used internally we could probably drop it and memoize `page_views`, `problem_reports`, `searches`, and `search_terms` directly.  It should reduce some of the indirection in the methods that use it.